tests/phpunit/unit-tests/models/class-llms-test-model-llms-membership.php

```diff
@@ -68,6 +68,73 @@ protected function get_data() {
 		);
 	}
 
+	/**
+	 * Test get_associated_posts() when none exist for the membership.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_get_associated_posts_none_found() {
+
+		$membership = $this->factory->membership->create_and_get();
+
+		$expected = array(
+			'post'   => array(),
+			'page'   => array(),
+			'course' => array(),
+		);
+		$this->assertEquals( $expected, $membership->get_associated_posts() );
+
+		$this->assertEquals( array(), $membership->get_associated_posts( 'course' ) );
+		$this->assertEquals( array(), $membership->get_associated_posts( 'page' ) );
+		$this->assertEquals( array(), $membership->get_associated_posts( 'post' ) );
+		$this->assertEquals( array(), $membership->get_associated_posts( 'fake' ) );
+
+	}
+
+	/**
+	 * Test get_associated_posts() when associations do exist.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_get_associated_posts_has_associations() {
+
+		$membership = $this->factory->membership->create_and_get();
+
+		// Add a post.
+		$post = $this->factory->post->create();
+		update_post_meta( $post, '_llms_is_restricted', 'yes' );
+		update_post_meta( $post, '_llms_restricted_levels', array( $membership->get( 'id' ), 1, 1008, '183' ) );
```

Comment:
how mean are you to stress my regexp with mixed types?

Reply:
:-p